### PR TITLE
Rework pod management

### PR DESCRIPTION
**Pull request by mook-as**
This implements the active/passive work.

Note that `mysql-proxy` isn't actually active/passive; it _should_ be, but it _actually_ is just enabled everywhere because [switchboard](https://github.com/cloudfoundry-incubator/switchboard/) doesn't implement any locking.

### CODE REVIEW

File: container-host-files/etc/scf/config/role-manifest.yml [hunk 1]
```diff
@@ -654,6 +680,8 @@ roles:
   processes:
   - name: tcp_router
   - name: metron_agent
+  tags:
+  - headless
```
[the remark this answers is not in the document]
Comment (mook-as) [reply container-host-files/etc/scf/config/role-manifest.yml]: You're correct, and we probably _should_ add that validation.

[the remark this answers is not in the document]
Comment (mook-as) [reply container-host-files/etc/scf/config/role-manifest.yml]: Validation added in https://github.com/SUSE/fissile/pull/354/commits/46102cb3f72cd1829f8722ea1c98992610e03517 — will update this PR in a bit

[the remark this answers is not in the document]
Comment (mook-as) [reply container-host-files/etc/scf/config/role-manifest.yml]: Role manifest fixed in 09f9311bb37c838e7b285225c4b4253e3afb6707 (which will be rebase into nothingness later)

**Comment by alexjh** [comment]
I'm seeing the cc-worker role in a non-ready state because of the rsyslog timestamp problem. I don't have a develop HEAD cluster to see what used to be logged to that file. I'm going to ask around to see if anyone has logs. @mook-as did you ever leave your cluster up for a while and notice this?

**Comment by mook-as** [comment]
Ah, [cc-worker](https://github.com/SUSE/scf/blob/de8212e34469eeb0d6bc38eaf5d6e92b0fef1b9e/container-host-files/etc/scf/config/role-manifest.yml#L834) is missing [syslog fixes](https://github.com/SUSE/scf/blob/de8212e34469eeb0d6bc38eaf5d6e92b0fef1b9e/container-host-files/etc/scf/config/role-manifest.yml#L965).  We should probably go through roles (in a separate PR please) to fix that.

As a workaround, you can edit the file in-place and `monit reload` to bring it back online.

**Comment by alexjh** [comment]
Should all pods have the syslog patch? I'm happy to blindly add it to all of the roles in a separate PR, but I don't know if it's meant to work that way.

**Comment by mook-as** [comment]
Yes, because we don't generate as much stuff in syslog as BOSH-based CF (where they run more things per VM).

**Comment by viovanov** [comment]
My manual tests show syslog draining working.
Also https://jenkins-new.howdoi.website/job/scf/view/change-requests/job/PR-1552/15/ passed.

**Pull request merged**